### PR TITLE
Implement reachable for inline assembly

### PR DESCRIPTION
**Pull request by LucasSte**
This PR implements the `reachable` function for the inline assembly statement.

### CODE REVIEW

File: src/sema/ast.rs [hunk 1]
```diff
@@ -1526,18 +1526,21 @@ impl Statement {
...
+
             Statement::If(_, reachable, ..)
             | Statement::While(_, reachable, ..)
             | Statement::DoWhile(_, reachable, ..)
-            | Statement::Expression(_, reachable, _) => *reachable,
-            Statement::Emit { .. } => true,
-            Statement::Delete(..) => true,
-            Statement::Continue(_) | Statement::Break(_) | Statement::Return(..) => false,
-            Statement::For { reachable, .. } | Statement::TryCatch(_, reachable, _) => *reachable,
-            Statement::Assembly(_) => unimplemented!("Assembly block ast not ready"),
```
Comment (LucasSte) [inline src/sema/ast.rs:1540]: @seanyoung This PR remove the `unimplemented!` from here. I haven't added any test.

Comment (seanyoung) [reply src/sema/ast.rs:1540]: Why no test?

Comment (LucasSte) [reply src/sema/ast.rs:1540]: Presently there is no test for the `reachable` function. I can create a unity test and test it altogether. Should I do so?

Comment (seanyoung) [reply src/sema/ast.rs:1540]: There are tests for unreachable statements: tests/contract_testcases/substrate/functions/for_forever.sol and tests/contract_testcases/substrate/loops/test_infinite_loop.sol

That's not great coverage, but best to improve this by adding tests for solidity statements after assembly statements.

Comment (LucasSte) [reply src/sema/ast.rs:1540]: I've included a contract test case and a unity test.

**Review by seanyoung — APPROVED** [review]

**Pull request merged**